include/cantera/kinetics/ReactionData.h

```diff
@@ -65,43 +89,29 @@ struct ArrheniusData
  * The data container `BlowersMaselData` holds precalculated data common to
  * all `BlowersMaselRate` objects.
  */
-struct BlowersMaselData
+struct BlowersMaselData final : public ReactionData
```

Comment:
I don't think marking these classes as `final` will have any appreciable performance impact, since the `update(bulk, kin)` function is only called once per rate type rather than once per reaction.